src/js/Core/Utils/TimeUtils.ts

```diff
@@ -75,8 +61,8 @@ export default class TimeUtils {
         return new Timer(duration);
     }
 
-    static resettableTimer(onDone: () => void|Promise<void>, duration: number): ResettableTimer {
-        return new ResettableTimer(onDone, duration);
+    static resettableTimer(callback: () => void|Promise<void>, duration: number): ResettableTimer {
```

Comment:
I thought to make it clear this is different than promise-based onDone/onCatch, since it is techically a callback that's scheduled to run after a duration 🤔

Reply:
Aren't promises using resolve, reject?